src/AbstractPropelSqlExtractor.php

```diff
@@ -0,0 +1,211 @@
+     *               The keys of the returned array are {@link K_UP_SQL} and {@link K_DOWN_SQL}.
+     */
+    protected function _extract($migration)
+    {
+        $code = $this->_getMigrationClassCode(get_class($migration));
+
+        if ($code === null) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not a PropelMigration object'),
+                null,
+                null,
+                $migration
+            );
+        }
+
+        $upSql = $this->_filterMigrationSqlQueries(
+            $migration->getUpSql(),
+            I::UP_MIGRATION,
+            $migration,
+            $code
+        );
+
+        $downSql = $this->_filterMigrationSqlQueries(
+            $migration->getDownSql(),
+            I::DOWN_MIGRATION,
+            $migration,
+            $code
+        );
+
+        $this->_finishExtraction($upSql, $downSql, $migration, $code);
+
+        return [
+            I::UP_MIGRATION   => $upSql,
+            I::DOWN_MIGRATION => $downSql,
+        ];
+    }
+
+    /**
+     * Retrieves the code of the migration class.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $className The migration class name.
+     *
+     * @return string|null The migration code or null if the code could not be determined.
+     */
+    protected function _getMigrationClassCode($className)
+    {
+        $matches = [];
+
+        preg_match('/PropelMigration_([0-9]*)$/', $className, $matches);
+
+        return count($matches) > 1
+            ? $matches[1]
+            : null;
+    }
+
+    /**
+     * Filters the migration SQL, potentially make changes or excluding some queries.
+     *
+     * @since [*next-version*]
+     *
+     * @param string[]|Stringable[] $sqlArray  An array of SQL queries.
+     * @param string|Stringable     $direction The direction of the query, either "up" or "down".
+     * @param object                $migration The migration instance.
+     * @param string|Stringable     $code      The migration code.
+     *
+     * @return string[]|Stringable[] The filtered migration queries, keyed by schema ID.
+     */
+    protected function _filterMigrationSqlQueries($sqlArray, $direction, $migration, $code)
+    {
+        $result = [];
+
+        foreach ($sqlArray as $_key => $_sql) {
+            $_filtered = $this->_filterMigrationSqlQuery($_sql, $_key, $direction, $migration, $code);
+
+            if ($_filtered !== null) {
+                $result[$_key] = $this->_normalizeString($_filtered);
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Normalizes the given direction string or string-like object.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $direction The direction.
+     *
+     * @return string The normalized direction.
+     */
+    protected function _normalizeDirection($direction)
```

Comment:
Seems to be unused in this class. Also, is implemented in `NormalizeDirectionCapableTrait`.